substrate/client/cli/src/commands/run_cmd.rs

```diff
@@ -345,7 +346,17 @@ impl CliConfiguration for RunCmd {
 	}
 
 	fn network_params(&self) -> Option<&NetworkParams> {
-		Some(&self.network_params)
+		let network_params = &self.network_params;
+		let is_authority = self.role(self.is_dev().ok()?).ok()?.is_authority();
+		if is_authority && network_params.public_addr.is_empty() {
+			warn!(
```

Comment:
nit: Could you please double check we don't silence this error? (ie start a local node validator without this addr config)

We may initialize the logger after logging some messages, and a few components of the CLI use the `eprintln!()` macro for logging for that purpose

Reply:
Good catch, updated